src/components/ColorPicker.tsx

```diff
@@ -26,17 +35,18 @@ class ColorPicker extends PureComponent {
         });
     };
 
-    handleClose = (event) => {
+    handleClose = (event: Event) => {
         if (event.stopPropagation) event.stopPropagation();
         this.setState({
             anchorEl: null,
         });
     };
 
-    handleColorChange = (color) => {
+    handleColorChange = (color: {hex: string}) => {
         this.setState({ color: color.hex }, () => {
-            if (this.props.isCustomEvent) changeCustomEventColor(this.props.customEventID, this.state.color);
-            else changeCourseColor(this.props.sectionCode, this.state.color, this.props.term);
+            // The && here is to keep the TS compiler happy. If isCustomEvent is true, there should always be a customEventID passed to props.
+            if (this.props.isCustomEvent && this.props.customEventID) changeCustomEventColor(this.props.customEventID, this.state.color);
+            else if(this.props.sectionCode&&this.props.term) changeCourseColor(this.props.sectionCode, this.state.color, this.props.term);
```

Comment:
Super nit: the spacing with &&

Reply:
I'm just gonna leave this untouched and let it fix itself when we auto-format everything.